### ISSUE TITLE
equalIgnoreCase fails with input including "... [a-z]"

### ISSUE THREAD

**Issue by niccottrell** [MorphiaOrg/morphia#1027]
Error like: 

```
java.util.regex.PatternSyntaxException: Illegal character range near index 223
^Third International Forum of  Santo Domingo  \u201CTime for reforms\u201D: What should Latin America do to become a more democratic, prosperous, equitable and safe region? [http://www.idea.int/americas/third-international-forum-of-santo-domingo-time-for-reforms.cfm]$
                                                                                                                                                                                                                               ^
    at java.util.regex.Pattern.error(Pattern.java:1955)
    at java.util.regex.Pattern.range(Pattern.java:2655)
    at java.util.regex.Pattern.clazz(Pattern.java:2562)
    at java.util.regex.Pattern.sequence(Pattern.java:2063)
    at java.util.regex.Pattern.expr(Pattern.java:1996)
    at java.util.regex.Pattern.compile(Pattern.java:1696)
    at java.util.regex.Pattern.<init>(Pattern.java:1351)
    at java.util.regex.Pattern.compile(Pattern.java:1054)
    at org.mongodb.morphia.query.FieldEndImpl.equalIgnoreCase(FieldEndImpl.java:91)
    at org.mongodb.morphia.query.FieldEndImpl.equalIgnoreCase(FieldEndImpl.java:25)
```

It seems that all instances of `Pattern.compile(x, Pattern.CASE_INSENSITIVE)` should be replaced with `Pattern.compile(string, Pattern.CASE_INSENSITIVE | Pattern.LITERAL)`, e.g. containsIgnoreCase, endsWithIgnoreCase, equalsIgnoreCase, startsWithIgnoreCase

**Comment by evanchooly** [comment]
This might be related to #853.  Can you attach the test case you're using?

**Comment by evanchooly** [comment]
I've added a test case that at least approximates what you have ( I hope ) and see the same error.  I added the `LITERAL` mask and it indeed fixes that case.  I'll clean up the other uses as appropriate and include the fix in 1.5.0.  Thanks for the heads up.  Apologies for the delay.  :)

**Comment by niccottrell** [comment]
No worries. I should have fixed it myself. I’m going to try and find the time to do some pull requests
